pkg/cri/resource-manager/requests.go

```diff
@@ -236,6 +236,7 @@ func (m *resmgr) RunPod(ctx context.Context, method string, request interface{},
 
 	pod := m.cache.InsertPod(podID, request, nil)
 	m.updateIntrospection()
+	m.updatePolicyMetrics()
```

Comment:
In my opinion, it is not a good idea to trigger (and wait) for a full collection cycle with every single CRI operation we intercept.

Granted we do the same for introspection data generation, but there is a huge difference. That one is a simple conversion of existing data already present in the policy to a canonical format. It does not need to reach out and collect an unknown amount of data from external sources of unknown latency. The `podpools` policy collector here will loop trough a bunch of cgroups and other data in `procfs`. We wait for that synchronously here and that adds that much latency to the processing of every request.

This comes to another point with the original design sketch up. Although it was not explicitly state there, our idea was to decouple policy-specific/`podpools` metrics collection from polling. IOW, to use an independent asynchronous goroutine within the (`podpools`) policy to periodically collect metrics with a configurable frequency/sampling rate and cache it internally. Then `podpools.PollMetrics()` would simply (read-lock the podpools policy) and return the already collected metrics data.

This would also have the additional benefit that your metrics sampling rate would be independent of the prometheus collection/sampling rate which would be a very good thing. Otherwise if you try to collect some more complex statistics you'll miss things as the externally clocked/triggered (relatively seldom) collection sampling rate will act as a low-pass filter on your data. As a stupid artificial example,  for instance if you tried to collect info about the CPU-'burstiness' of containers in a pool, then with a 10 second externally triggered collection you miss containers that steadily alternate between 1 second of extreme load and 1 second of idleness.